app/lib/features/news/widgets/news_item_slide/image_slide.dart

```diff
@@ -5,56 +5,64 @@ import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:logging/logging.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
 
 final _log = Logger('a3::news::widget::image_slide');
 
 class ImageSlide extends StatelessWidget {
   final NewsSlide slide;
-  final Color bgColor;
-  final Color fgColor;
 
-  const ImageSlide({
-    super.key,
-    required this.slide,
-    required this.bgColor,
-    required this.fgColor,
-  });
+  const ImageSlide({super.key, required this.slide});
 
   @override
   Widget build(BuildContext context) {
+    return Container(
+      key: NewsUpdateKeys.imageUpdateContent,
+      child: renderImageContent(),
+    );
+  }
+
+  Widget renderImageContent() {
     return FutureBuilder<FfiBufferUint8>(
       future: slide.sourceBinary(null),
       builder: (BuildContext context, AsyncSnapshot<FfiBufferUint8> snapshot) {
-        if (snapshot.hasError) {
-          _log.severe(
-            'Failed to load image of slide',
-            snapshot.error,
-            snapshot.stackTrace,
-          );
-          return Center(
-            child: Text(L10n.of(context).errorLoadingImage(snapshot.error!)),
-          );
-        }
-
         if (snapshot.hasData &&
             snapshot.connectionState == ConnectionState.done) {
-          return Container(
-            color: bgColor,
-            alignment: Alignment.center,
-            key: NewsUpdateKeys.imageUpdateContent,
-            foregroundDecoration: BoxDecoration(
-              image: DecorationImage(
-                fit: BoxFit.contain,
-                image: MemoryImage(
-                  Uint8List.fromList(snapshot.data!.asTypedList()),
-                ),
-              ),
-            ),
-          );
+          return buildImageUI(snapshot.data!.asTypedList());
+        } else if (snapshot.hasError) {
+          return buildImageLoadingErrorUI(context, snapshot);
         }
-
-        return Center(child: Text(L10n.of(context).loadingImage));
+        return buildImageLoadingUI();
       },
     );
   }
+
+  Widget buildImageUI(Uint8List imageData) {
```

Comment:
I feel like chat uses same layout for image rendering, this could be simplified to be reusable ?

Reply:
`Reusable` you are referring is just a simple `container` with 4-5 lines of code. In chat image there bit different shape and UI requirement `image_message_builder.dart`. So for of 4-5 lines of code, I would prefer to keep it separate to reduce complexity.